### PR TITLE
Added model support to test-maxtext.sh

### PR DESCRIPTION
**Pull request by kocchop**
Added more functionality to the test script:
1. added more model support
2. added parallelism heuristic
3. added fp8 test support 
4. added example launch commands

### CODE REVIEW

File: .github/container/test-maxtext.sh [hunk 1]
```diff
@@ -13,26 +13,40 @@ usage() {
...
-    echo "  --dtype                    Batch size, defaults to bfloat16."
+    echo "  --dtype                    Data type, defaults to bfloat16. Example: bfloat16, fp8"
     echo "  -s, --steps                Number of steps to run, defaults to 500."
-    echo "  --multiprocess             Enable the multiprocess GPU mode."
-    echo "  -o, --output NAME          Name for the output folder, a temporary folder will be created if none specified."
-    echo "  --data-parallel            Data parallelism to use. Defaults to 1."
-    echo "  --fsdp                     Fully-sharded data parallelism to use. Defaults to 1."
-    echo "  --tensor-parallel          Tensor parallelism to use. Defaults to 1."
+    echo "  --multiprocess             Enable the multiprocess GPU mode. Should be used when run on multinode"
+    echo "  -o, --output               Name for the output folder, a temporary folder will be created if none specified."
```
Comment (nouiz) [inline .github/container/test-maxtext.sh:27]: Maybe add back NAME?

File: .github/container/test-maxtext.sh [hunk 2]
```diff
@@ -139,26 +158,39 @@ done
 
 # # Set derived variables
 
+# if not the gpt3-52k, we can use any any attention type such as cudnn_flash_te or dot_product
+if [ $MODEL != "gpt3-52k" ]; then # gpt3-52k only works with dot_product
+    ADDITIONAL_ARGS+=" attention=${ATTN_TYPE}"
+fi
+
+# for fp8 runs
+if [ $DTYPE == "fp8" ]; then # gpt3-52k only works with dot_product
```
Comment (nouiz) [inline .github/container/test-maxtext.sh:167]: remove the comment?

File: .github/container/test-maxtext.sh [hunk 3]
```diff
@@ -139,26 +158,39 @@ done
...
-if [ $NGPUS -gt 8 ]; then
-    dcn_DP=$((NGPUS / 8))
-    ici_DP=$((DP / dcn_DP))
+# TP is always ici; after TP it will be FSDP and
+# from TP and FSDP, we can find out ici and dcn DP
+# in other words, DP dim across ici and dcn axis will always be inferred
+ici_TP=${TP}
+ici_DP=1
+dcn_FSDP=1
+if [ $((FSDP*TP)) -gt 8 ]; then
```
Comment (nouiz) [inline .github/container/test-maxtext.sh:181]: Maybe add a (c) note that the script expect 8 GPUs per node.
If needed, later someone can lift.

File: .github/container/test-maxtext.sh [hunk 4]
```diff
@@ -191,7 +224,7 @@ export BASE_XLA_FLAGS=${BASE_XLA_FLAGS:---xla_gpu_enable_latency_hiding_schedule
                 --xla_gpu_enable_highest_priority_async_stream=true
                 --xla_gpu_all_reduce_combine_threshold_bytes=1073741824 
                 --xla_gpu_all_gather_combine_threshold_bytes=1073741824 
-                --xla_gpu_reduce_scatter_combine_threshold_bytes=1073741824
+                --xla_gpu_reduce_scatter_combine_threshold_bytes=134217728
```
Comment (nouiz) [inline .github/container/test-maxtext.sh:227]: Why change this in this PR?

**Review by nouiz — COMMENTED** [review]
Small comments only.

File: .github/container/test-maxtext.sh (hunk 4 above)
Comment (kocchop) [reply .github/container/test-maxtext.sh:227]: For RS, the combiner threshold represents the output buffer size. Typically, the `msg size=output_buffer_size*num_gpus`. Hence, to get an effective RS msg size of 1G, we can nominally set it to 128MB assuming 8 gpus on 1 node. For multinode runs, all these values need to be tuned accordingly for best perf.

File: .github/container/test-maxtext.sh (hunk 3 above)
Comment (kocchop) [reply .github/container/test-maxtext.sh:181]: good point, I can actually use the variable `GPUS_PER_NODE=$(nvidia-smi -L | grep -c '^GPU')` then it should be alright

File: .github/container/test-maxtext.sh (hunk 2 above)
Comment (kocchop) [reply .github/container/test-maxtext.sh:167]: sure, sorry I missed that

File: .github/container/test-maxtext.sh (hunk 1 above)
Comment (kocchop) [reply .github/container/test-maxtext.sh:27]: sure, will add it back

File: .github/container/test-maxtext.sh (hunk 4 above)
Comment (nouiz) [reply .github/container/test-maxtext.sh:227]: Ok. So this values is a better default value. But it still need to be tuned.

File: .github/container/test-maxtext.sh [hunk 5]
```diff
@@ -13,26 +13,40 @@ usage() {
     echo "Usage: $0 [OPTIONS]"
     echo ""
     echo "  OPTIONS                    DESCRIPTION"
-    echo "  -a, --additional-args      Additional fiddle args to pass to MaxText/train.py"
+    echo "  -a, --additional-args      Additional args to pass to MaxText/train.py"
     echo "  --mem-fraction             Specify the percentage of memory to preallocate for XLA. Example: 0.90, 0.85, 0.65"
-    echo "  --decoder-block            Specify decoder block to run. Example: llama2, default"
-    echo "  --attn-type                Specify the attention type. Example: dot_product, cudnn_flash_te"
+    echo "  --model-name               Specify the model names to run [Preffered]. If you specify model name then you do not need to specify decoder-block. Currently supported ootb models:
```
Comment (nouiz) [inline .github/container/test-maxtext.sh:18]:
```suggestion
    echo "  --model-name               Specify the model names to run [Preferred]. If you specify model name then you do not need to specify decoder-block. Currently supported ootb models: 
```

**Review by nouiz — APPROVED** [review]

File: .github/container/test-maxtext.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (kocchop) [reply .github/container/test-maxtext.sh]: Thanks for noticing.. Its fixed now

**Review by sergachev — APPROVED** [review]

**Pull request merged**